src/main/app/src/components/hakutulos/hakutulosSuodattimet/KoulutusalatSuodatin.js

```diff
@@ -139,68 +141,67 @@ const KoulutusalatSuodatin = observer(() => {
-                      {_.isNil(
-                        expandedKoulutusTaso1[1]?.alakoodit?.[koulutusTaso2_ID]
-                          ?.count
-                      )
-                        ? ''
-                        : `(${expandedKoulutusTaso1[1]?.alakoodit?.[koulutusTaso2_ID]?.count})`}
-                    </Grid>
-                  </Grid>
+        Object.keys(expandedKoulutusTaso1[1]?.alakoodit).map((koulutusTaso2_ID) => (
+          <ListItem
+            key={koulutusTaso2_ID}
+            dense
+            button
+            onClick={handleKoulutusalaInnerToggle(
+              koulutusTaso2_ID,
+              expandedKoulutusTaso1[1]?.alakoodit?.[koulutusTaso2_ID]
+            )}
+            disabled={
+              expandedKoulutusTaso1[1]?.alakoodit?.[koulutusTaso2_ID]?.count === 0
+            }>
+            <ListItemIcon>
+              <SuodatinCheckbox
+                edge="start"
+                checked={
+                  valitutKoulutusAlat.findIndex(({ id }) => id === koulutusTaso2_ID) !==
+                  -1
                 }
+                tabIndex={-1}
+                disableRipple
               />
-            </ListItem>
-          )
-        )}
+            </ListItemIcon>
+            <SuodatinListItemText
+              id={`${expandedKoulutusTaso1[0]}_${koulutusTaso2_ID}`}
+              primary={
+                <Grid container justify="space-between" wrap="nowrap">
+                  <Grid item>
+                    {
+                      expandedKoulutusTaso1[1]?.alakoodit?.[koulutusTaso2_ID]?.nimi?.[
+                        i18n.language
+                      ]
+                    }
+                  </Grid>
+                  <Grid item>
+                    {_.isNil(
+                      expandedKoulutusTaso1[1]?.alakoodit?.[koulutusTaso2_ID]?.count
+                    )
+                      ? ''
+                      : `(${expandedKoulutusTaso1[1]?.alakoodit?.[koulutusTaso2_ID]?.count})`}
+                  </Grid>
+                </Grid>
+              }
+            />
+          </ListItem>
+        ))}
     </List>
   );
 
   return (
-    <SuodatinExpansionPanel defaultExpanded={true}>
+    <SuodatinExpansionPanel elevation={elevation} defaultExpanded={expanded}>
       <SuodatinExpansionPanelSummary expandIcon={<ExpandMore />}>
-        <Typography variant="subtitle1">{t('haku.koulutusalat')}</Typography>
+        <SummaryContent
+          selectedFiltersStr={selectedKoulutusalatStr}
+          maxCharLengthBeforeChipWithNumber={20}
+          filterName={t('haku.koulutusalat')}
+          displaySelected={displaySelected}
+        />
       </SuodatinExpansionPanelSummary>
       <SuodatinExpansionPanelDetails>
-        <List
-          hidden={expandedKoulutusTaso1.length > 0}
-          style={{ width: '100%' }}>
+        <List hidden={expandedKoulutusTaso1.length > 0} style={{ width: '100%' }}>
           {koulutusAlat.map((kouutusalaArray) => {
```

Comment:
Tässä on näköjään vielä jäänyt pieni typo
```suggestion
          {koulutusAlat.map((koulutusalaArray) => {
```